internal/io/csv.go

```diff
@@ -98,6 +101,27 @@ func ReadCSV(reader io.Reader, conf CSVConfig) (map[string]interface{}, []string
 		}
 	}
 
+	if conf.MissingColumnNameAlias != "" {
+		for i, name := range headers {
+			if name == "" {
+				headers[i] = conf.MissingColumnNameAlias
+			}
+		}
+	}
+
+	if conf.AddColumnIndexToDuplicateColumns {
+		headerSet := strings.NewEmptyStringSet()
+		for i, h := range headers {
+			if headerSet.Contains(h + fmt.Sprint(i)) {
```

Comment:
What would happen for ´COL1,COL,COL11´ (which is valid as is) for example?

I think you somehow need to take into consideration not only the headers that you've looped over but also those that come later since you don't want to create conflicts with existing column names. Perhaps a ´map[string]int´ mapping the column name to the first column position of the name  that you populate initially would be a better data structure for this than the stringset?

Reply:
I knew this was going to be issue, I was thinking through something. might be able to test it next commit.